src/main/java/org/truffleruby/core/range/RangeNodes.java

```diff
@@ -76,6 +77,13 @@ public DynamicObject map(DynamicObject range, DynamicObject block,
             return createArray(arrayBuilder.finish(store, length), length);
         }
 
+        @Specialization(guards = "isObjectRange(range)")
+        public Object mapObject(VirtualFrame frame, DynamicObject range, DynamicObject block,
+                                @Cached("new()") SnippetNode snippetNode) {
+            return snippetNode.execute(frame, "ret = []; range.each { |x| ret << block.call(x) }; ret",
+                    "range", range, "block", block);
+        }
```

Comment:
Could you make this a primitive and have a fallback specialization so this code is not hidden in a Snippet but lives in core/range.rb? This should also handle `long` ranges trivially.

Reply:
Check 5232f8afd56b3e24fec71b20342b7024e6732811 and see if that's what you had in mind, please.

Reply:
Yes it is, thanks.